fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneIndexTest.java

```diff
@@ -1352,6 +1359,180 @@ void simpleCrossPartitionQuery() {
+                        List.of(new LuceneQueryMultiFieldSearchClause(LuceneQueryType.QUERY, luceneSearch, false)
+                        )
+                                                                      :
+                        List.of(new LuceneQueryMultiFieldSearchClause(LuceneQueryType.QUERY, luceneSearch, false),
+                                new LuceneQueryFieldComparisonClause.LongQuery(
+                                        LuceneQueryType.QUERY, isSynthetic ? "complex_timestamp" : "timestamp",
+                                        LuceneIndexExpressions.DocumentFieldType.LONG,
+                                        new Comparisons.SimpleComparison(comparisonType, predicateComparand),
+                                        false,
+                                        null)
+                        );
+                LuceneQueryClause clause = new LuceneBooleanQuery(LuceneQueryType.QUERY, luceneQueryClauses, BooleanClause.Occur.SHOULD);
+
+                LuceneScanQueryParameters scan = new LuceneScanQueryParameters(
+                        Verify.verifyNotNull(ScanComparisons.from(new Comparisons.SimpleComparison(Comparisons.Type.EQUALS, 1))),
+                        clause,
+                        sort,
+                        null,
+                        null,
+                        null);
+                return scan.bind(recordStore, index, EvaluationContext.EMPTY);
+            }
+        }
+
+        LuceneQueryBuilder luceneQueryBuilder = new LuceneQueryBuilder();
+        try (FDBRecordContext context = openContext(contextProps)) {
+            schemaSetup.accept(context);
+            LucenePartitioner partitioner = getIndexMaintainer(index).getPartitioner();
+            Tuple groupKey = Tuple.from(1L);
+            
+            // timestamps present in partition keys
+            long time0 = System.currentTimeMillis();
+            long time1 = System.currentTimeMillis() + 1000;
+            long time2 = System.currentTimeMillis() + 2000;
+            
+            // timestamp not in partition keys, but within a partition
+            long time1_2 = time1 + 500; // between time1 and time2
+
+            // timestamps outside the bounds of the partitions
+            long timeTooOld = time0 - 500;
+            long timeTooNew = time2 + 500;
+
+            Map<Long, String> timesForLogging = Map.of(
+                    time0, "time0",
+                    time1, "time1",
+                    time2, "time2",
+                    time1_2, "time1_2",
+                    timeTooOld, "timeTooOld",
+                    timeTooNew, "timeTooNew");
+
+            //
+            // p0: from: t0+1200 to: t0+1300
+            // p1: from: t0+1301 to: t1+1400
+            // p2: from: t1+1410 to: t2+1500
+            // p3: from: t2+1510 to: t2+1600
+            //
+            createPartitionMetadata(index, groupKey, 0, time0, time0, Tuple.from(1, 1200), Tuple.from(1, 1300));
```

Comment:
```suggestion
            // create partition metadata directly rather than depending on repartitioning to ensure that the partitions
            // are what we want
            createPartitionMetadata(index, groupKey, 0, time0, time0, Tuple.from(1, 1200), Tuple.from(1, 1300));
```